File: tests/LargeCaseRange.cs
namespace N {
    public partial class TRange {
        public void Foo(int x) {
            switch (x)
            {
                case var v when v >= 1 && v <= 1005: System.Console.WriteLine("range"); break;
                case 2000: System.Console.WriteLine("other"); break;
            }
        }
    }
}

File: tests/LargeCaseRange.pas
namespace N;

type
  TRange = public class
  public
    method Foo(x: Integer);
  end;

implementation

method TRange.Foo(x: Integer);
begin
  case x of
    1..1005: System.Console.WriteLine('range');
    2000: System.Console.WriteLine('other');
  end;
end;

end.
